components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultClaimProvider.java

```diff
@@ -0,0 +1,151 @@
+                                                   OAuth2AuthorizeRespDTO authorizeRespDTO)
+            throws IdentityOAuth2Exception {
+
+        Map<String, Object> claims = new HashMap<>();
+        String[] cachedRequests = null;
+        final String sessionDataKey = authAuthzReqMessageContext.getAuthorizationReqDTO().getSessionDataKey();
+        if (StringUtils.isNotBlank(sessionDataKey)) {
+            cachedRequests = SessionDataCache.getInstance()
+                    .getValueFromCache(new SessionDataCacheKey(sessionDataKey)).getParamMap().get("request");
+        }
+        if (cachedRequests != null && !(cachedRequests[0].split(JWT_PART_DELIMITER).length == NUMBER_OF_PARTS_IN_JWE)) {
+            JSONObject requestBody = getRequestBodyFromCache(cachedRequests);
+
+            /* State is an optional parameter, so the authorization server must successfully authenticate and
+             * must NOT return state nor s_hash. (FAPI1-ADV-5.2.2.1-5)
+             */
+            final String state = requestBody.getAsString(OAuthConstants.OAuth20Params.STATE);
+            if (StringUtils.isNotEmpty(state)) {
+                claims.put(IdentityCommonConstants.S_HASH, IdentityCommonUtils.getHashValue(state, null));
+            } else {
+                // state is empty, removing state from cache too
+                removeStateFromCache(sessionDataKey);
+            }
+        }
+        final String responseType = authAuthzReqMessageContext.getAuthorizationReqDTO().getResponseType();
+        avoidSettingATHash(responseType, authorizeRespDTO, claims);
+
+        return claims;
+
+    }
+
+    @Override
+    public Map<String, Object> getAdditionalClaims(OAuthTokenReqMessageContext tokenReqMessageContext,
+                                                   OAuth2AccessTokenRespDTO tokenRespDTO)
+            throws IdentityOAuth2Exception {
+
+        return new HashMap<>();
+    }
+
+    /**
+     * If response_type value is not 'code id_token token', avoid setting at_hash claim to the authorization
+     * endpoint id_token as it is OPTIONAL (OIDCC-3.3.2.11).
+     *
+     * @param responseType     requested auth response_type
+     * @param authorizeRespDTO authorizeRespDTO
+     * @param claims           returning claims map
+     */
+    private void avoidSettingATHash(String responseType, OAuth2AuthorizeRespDTO authorizeRespDTO,
+                                    Map<String, Object> claims) {
+
+        if (StringUtils.isNotBlank(responseType)) {
+            List<String> responseTypes = Arrays.asList(responseType.trim().split("\\s+"));
+            if (!(responseTypes.contains(IdentityCommonConstants.CODE)
+                    && responseTypes.contains(OAuthConstants.ID_TOKEN)
+                    && responseTypes.contains(OAuthConstants.TOKEN))) {
+                if (StringUtils.isNotBlank(authorizeRespDTO.getAccessToken())) {
+                    authorizeRespDTO.setAccessToken(null);
+                }
```

Comment:
This part is not added in the method comment, shall we add it to the method comment for clarity? Hope checking for the presence of accesstoken and setting it to null if present is a valid scenario.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136